pkg/haproxy/haproxy.go

```diff
@@ -291,6 +291,30 @@ func haproxyContainer(cr *apiv1alpha1.PerconaServerMySQL) corev1.Container {
 	}
 	env = append(env, spec.Env...)
 
+	var readinessProbe, livenessProbe *corev1.Probe
+	if cr.CompareVersion("0.12.0") >= 0 {
+		readinessProbe = &spec.ReadinessProbe
+		readinessProbe.Exec = &corev1.ExecAction{
+			Command: []string{"/opt/percona/haproxy_readiness_check.sh"},
+		}
+		livenessProbe = &spec.LivenessProbe
+		livenessProbe.Exec = &corev1.ExecAction{
+			Command: []string{"/opt/percona/haproxy_liveness_check.sh"},
+		}
```

Comment:
i think we can use `k8s.ExecProbe` here

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/975/commits/e9eb524e7b58c8d10f96b77ab70eea6ca47f0003